cloud/edgecontroller/pkg/devicecontroller/types/deviceprofile.go

```diff
@@ -0,0 +1,52 @@
+package types
+
+// DeviceProfile is structure to store in configMap
+type DeviceProfile struct {
+	DeviceInstances  []*DeviceInstance  `json:"deviceInstances,omitempty"`
```

Comment:
please add comments for all the fields in the struct. these types would be consumed by mappers to marshal / unmarshal the device profile json .